### ISSUE TITLE
Support for ConversationView with existing application bottom tabbar

### ISSUE THREAD

**Issue by wireframe** [layerhq/Atlas-iOS#1064]
The current conversation view hardcodes the location of the chat input cell directly over the bottom tab bar.  We would like to use the atlas conversation view within our current view.

**Comment by kmahal** [comment]
@wireframe please give more details as it is unclear what you are specifying/requesting.

**Comment by wireframe** [comment]
Here's a screenshot from our integration.  We have standard bottom tab buttons for navigation and when you try to navigate to the "conversation tab", the atlas conversation view pins the text input to the bottom of the screen which covers up the tab navigation buttons at the bottom of the screen.
<img width="198" alt="iphone_6_-_iphone_6___ios_9_1__13b137__720" src="https://cloud.githubusercontent.com/assets/12001/11846736/84594cc0-a3df-11e5-94d1-d31b066571ac.png">

The current atlas conversation view component works well within a modal controller UX, but it seems very necessary to support using the conversation view controller outside of a modal flow and embed into a standard app navigation.

**Comment by kmahal** [comment]
@wireframe Atlas components cannot know what they are embedded in by default and work well with most standard app navigations, such as embedded in a navigation controller.  Your application can have the conversation list in the tabbar controller, and on presenting the `ATLConversationViewController` you should hide the bottom tab bar.  The `ATLMessageInputToolbar` is set as the view's input accessory view which is then managed by the application keyboard, which is why it's anchored to the bottom.  If you'd like to have the input toolbar's `origin.y` be something beyond what it is, you can do that yourself with a custom subclass of the toolbar and overriding `layoutSubviews`.

**Comment by wireframe** [comment]
> Atlas components cannot know what they are embedded in by default and work well with most standard app navigations

I would disagree that the current solution works well in *most* navigations.  Having the textarea overlay the navigation tabbar is a very non-standard UI interaction *unless* you are using a modal interaction.

Yes, subclassing is an option to override and fix this behavior, but it is a pretty heavy handed approach to solve this issue.  If the atlas library cannot infer it's display state, would it be possible to allow the caller to pass in an option into the controller when instantiated to configure the desired behavior?

If subclassing is the only option, feel free to close this issue out and we will continue to explore other options.

**Comment by r0ct0-tecsynt** [comment]
I face close problem. I have tabbarcontroller-.>navcontroller->conversationlistocntroler->push  conversation controller . If i tap on any textfield (address bar or message input) - they will ump and disappear, after that i can see tabor uncovered

**Comment by ancheng1114** [comment]
@wireframe 
can you please explain how to implement subclass in case of bottom tabbar?

**Comment by jessearmand** [comment]
I have the same problem #1081

**Comment by siskaj** [comment]
@kmahal 
I have the same problem. ATLConversationViewController does not work well the UItabbarController. I searched the WEB and I have not been able to find any simple solution.
Can we expect any solution from Atlas or do you plan to leave it as it is?

**Comment by jessearmand** [comment]
The simple solution is to hide UITabBar of the UITabBarController when the conversation view controller is pushed.
Alternatively you can create a container view controller and add that conversation view controller as a child view controller, and do more custom setup or layout configurations. 
But I'll take the first approach as it's less code, less dependency and less potential layout problems.

Sent from Outlook Mobile

    _____________________________

**Comment by siskaj** [comment]
@jessearmand 

Thank you Jesse,

I have created button and tried hiding/unhiding UITabBar and hiding/unhiding  messageInputToolbar and all apparently worked fine till sending a message. Then messageInputToolbar is not visible and most likely not the firstRespondent either.

The container solution seems to me quite complicated - layout configuration and positioning of keybord properly will be quite likely a lot of work.

**Comment by feifanzhou** [comment]
@siskaj Are you hiding both the toolbar and tabbar? Simply hide the tabbar on `viewDidAppear` and show it on `viewDidDisappear`. The toolbar should then remain onscreen all the time.

**Comment by kcoleman731** [comment]
@wireframe Apologies for letting this ticket go stale. For now we are going to recommend @jessearmand suggestion of hiding the tab bar. If you want to take a shot at a PR which supports your suggestion of passing an option to the controller for positioning the input tool bar, I'd be happy to have a look.

**Comment by bartoszhernas** [comment]
Hi,

I've found workaround, feel free to use my fork :)

https://github.com/whatsahoy/Atlas-iOS/commit/5c492ac61226bdbdf755537d41876def3ab26b47

**Comment by LukeDurrant** [comment]
See https://github.com/layerhq/Atlas-iOS/pull/1363

**Comment by SiarheiMinkin** [comment]
In subclass of ATLConversationViewController insert next code:
- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    CGRect frame = self.messageInputToolbar.superview.frame;
    frame.origin.y = self.view.frame.size.height - 40;
    self.messageInputToolbar.superview.frame = frame;
}

Works for me :)
